File: 06-bstpractice-Python/bst.py
class Node(object):
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

class BST(object):
    def __init__(self, root):
        self.root = Node(root)

    def insert(self, new_val):
        # Your code goes here
        if self.root == None:
            return newNode(new_val)
        
        if new_val < self.root.value:
            lchild = self.insert(self.root.left, new_val)
            root.left = lchild

        elif new_val > self.root.value:
            rchild = self.insert(self.root.right, new_val)
            root.right = rchild

        return self.root  


    def printSelf(self):
        # Your code goes here
        if self.left:
            self.left.printSelf()
        if self.right:
            self.right.printSelf()
        
    def search(self, find_val):
        # Your code goes here
        if self.root is None or self.root.value == find_val:
            return True
        return False
        if self.root.value < find_val:
            search(self.root.right, find_val)
        else:
            search(self.root.left, find_val)

File: 06-bstpractice-Java/BST.java

class Node {
    public int value;
    public Node left, right;
    public Node(int value) {
        this.value = value;
        this.left = null;
        this.right = null;
    }
}

public class BST {
    public Node root;
    
    public BST(int value) {
        this.root = new Node(value);
    }

    public void insert(int value) {
        // Your code goes here
        insert(root, value);
    }

    private void insert(Node node, int value) {
        // Your code goes here
        
        if(value < node.value) {

            if(node.left == null) {

                node.left = new Node(value);
                return;
            } else {

                insert(node.left, value);
            }
        } else {

            if(node.right == null) {

                node.right = new Node(value);
                return;
            } else {

                insert(node.right, value);
            }
        }
    }

    public boolean search(int value) {
    	// Your code goes here
    	if(root.value == value) {

            return true;
        }
        return search(root, value);
    }

    private boolean search(Node current, int value) {
    	// Your code goes here
        
        if(current == null) {

            return false;
        } else if (current.value == value) {

            return true;
        } else {

            if (value < current.value) {

                return search(current.left, value);
            } else {

                return search(current.right, value);
            }
        }
    }

}